src/solver/optimisation/opt_gestion_second_membre_cas_lineaire.cpp

```diff
@@ -31,206 +31,435 @@
-                AdresseOuPlacerLaValeurDesCoutsMarginaux[cnt]
-                  = problemeHebdo->ResultatsContraintesCouplantes[cntCouplante].variablesDuales.data()
-                    + pdtHebdo;
-            }
+            builder.updateHourWithinWeek(pdt);
+            builder.include(Variable::DispatchableProduction, poids, offset, true);
         }
     }
+    // TODO probably wrong from the 2nd week, check
+    const int jour = problemeHebdo->NumeroDeJourDuPasDeTemps[pdtDebut];
+    double rhs = MatriceDesContraintesCouplantes.SecondMembreDeLaContrainteCouplante[jour];
+    switch(MatriceDesContraintesCouplantes.SensDeLaContrainteCouplante) {
+    case '=':
+      builder.equal(rhs);
+      break;
+    case '<':
+      builder.less(rhs);
+      break;
+    case '>':
+      builder.greater(rhs);
+      break;
+      //TODO default case ?
+    }
+    {
+      ConstraintNamer namer(problemeHebdo->ProblemeAResoudre->NomDesContraintes, problemeHebdo->NamedProblems);
+      namer.UpdateTimeStep(jour);
+      namer.BindingConstraintDay(problemeHebdo->ProblemeAResoudre->NombreDeContraintes,
+                                 MatriceDesContraintesCouplantes.NomDeLaContrainteCouplante);
+    }
+    builder.build();
+    }
+  }
+};
+
+void OPT_BuildConstraints(PROBLEME_HEBDO* problemeHebdo,
+                          int PremierPdtDeLIntervalle,
+                          int DernierPdtDeLIntervalle,
+                          int NumeroDeLIntervalle,
+                          const int optimizationNumber)
+{
+    int weekFirstHour = problemeHebdo->weekInTheYear * 168;
+
+    auto& ProblemeAResoudre = problemeHebdo->ProblemeAResoudre;
+
+    std::vector<double>& SecondMembre = ProblemeAResoudre->SecondMembre;
 
-    for (int pdtHebdo = PremierPdtDeLIntervalle; pdtHebdo < DernierPdtDeLIntervalle;)
+    std::vector<double*>& AdresseOuPlacerLaValeurDesCoutsMarginaux
+      = ProblemeAResoudre->AdresseOuPlacerLaValeurDesCoutsMarginaux;
+
+    const std::vector<int>& NumeroDeJourDuPasDeTemps = problemeHebdo->NumeroDeJourDuPasDeTemps;
+
+    for (int i = 0; i < ProblemeAResoudre->NombreDeContraintes; i++)
     {
-        int jour = NumeroDeJourDuPasDeTemps[pdtHebdo];
-        int indexCorrespondanceCnt = (!problemeHebdo->OptimisationAuPasHebdomadaire) ? 0 : jour;
+        AdresseOuPlacerLaValeurDesCoutsMarginaux[i] = nullptr;
+        SecondMembre[i] = 0.0;
+    }
 
-        CORRESPONDANCES_DES_CONTRAINTES_JOURNALIERES& CorrespondanceCntNativesCntOptimJournalieres
-              = problemeHebdo->CorrespondanceCntNativesCntOptimJournalieres[indexCorrespondanceCnt];
+    // TODO reset selectively
+    ProblemeAResoudre->NombreDeContraintes = 0;
+    ProblemeAResoudre->NombreDeTermesDansLaMatriceDesContraintes = 0;
 
-        for (int cntCouplante = 0; cntCouplante < problemeHebdo->NombreDeContraintesCouplantes;
+    AreaBalance areaBalance(problemeHebdo);
+    FictiveLoad fictiveLoad(problemeHebdo);
```

Comment:
I am not sure anymore of the added value of the comon interface `IConstraint`: it does not seem to be used, we only use plain values instead?

Reply:
Yes, this interface can be safely removed. 

Fictitious seems better suited indeed.